robotoff/spellcheck/vocabulary/utils.py

```diff
@@ -0,0 +1,153 @@
+            self.deaccented_tokens = self.deaccent_tokens_fn(self.voc)()
+
+    def __contains__(self, token: str) -> bool:
+        return self.normalize(token) in self.voc
+
+    def _contains_deaccent(self, token: str) -> bool:
+        return self.deaccent(token) in self.deaccented_tokens
+
+    def __or__(self, other):
+        return Vocabulary(
+            tokens=self.voc | other.voc,
+            deaccented_tokens=defaultdict(
+                list, dict(self.deaccented_tokens, **other.deaccented_tokens)
+            ),
+        )
+
+    def suggest(self, token: str) -> Optional[str]:
+        deaccent_suggestions: Optional[List[str]] = self._suggest_deaccent(token)
+        if deaccent_suggestions is not None:
+            if len(deaccent_suggestions) == 1:
+                return deaccent_suggestions[0]
+            elif len(deaccent_suggestions) > 1:
+                return None
+
+        split_suggestions: List[Tuple[str, str]] = self._suggest_split(token)
+        if len(split_suggestions) == 1:
+            return split_suggestions[0][0] + " " + split_suggestions[0][1]
+        return None
+
+    def _suggest_deaccent(self, token: str) -> Optional[List[str]]:
+        if token in self:
+            return None
+        deaccented_token = self.deaccent(token)
+        if deaccented_token in self.deaccented_tokens:
+            return self.deaccented_tokens[deaccented_token]
+        return None
+
+    def _suggest_split(self, token: str) -> List[Tuple[str, str]]:
+        if token in self:
+            return []
+
+        suggestions = []
+        for i in range(2, len(token) - 1):
+            # pre and post must be at least 2 letters long
+            pre = token[:i]
+            post = token[i:]
+            if pre in self and post in self or (pre + " " + post) in self:
+                suggestions.append((pre, post))
+            else:
+                if self._contains_deaccent(pre) and post in self:
+                    pre_suggestion = self.deaccented_tokens[self.deaccent(pre)]
+                    if len(pre_suggestion) == 1:
+                        suggestions.append((pre_suggestion[0], post))
+
+                if pre in self and self._contains_deaccent(post):
+                    post_suggestion = self.deaccented_tokens[self.deaccent(post)]
+                    if len(post_suggestion) == 1:
+                        suggestions.append((pre, post_suggestion[0]))
+
+                if self._contains_deaccent(pre) and self._contains_deaccent(post):
+                    pre_suggestion = self.deaccented_tokens[self.deaccent(pre)]
+                    post_suggestion = self.deaccented_tokens[self.deaccent(post)]
+                    if len(pre_suggestion) == 1 and len(post_suggestion) == 1:
+                        suggestions.append((pre_suggestion[0], post_suggestion[0]))
+        return suggestions
+
+    @staticmethod
+    def deaccent(token: str) -> str:
+        ACCENTS = {"a": "à", "e": "éêè", "u": "ùüû", "i": "ïî"}
```

Comment:
I just realized that I left this code to deaccent words. This list is specific to french and is not even exhaustive. I'll change this by a proper package.